usr/src/cmd/acct/acctcom.c

```diff
@@ -556,7 +556,7 @@ println(void)
 	    a->ac_tty != 0xffff? devtolin(a->ac_tty):"?");
 #else
 	fprintf(stdout, " %-*.*s", OUTPUT_LSZ, OUTPUT_LSZ,
-	    a->ac_tty != (dev_t)-1? devtolin(a->ac_tty):"?");
+	    a->ac_tty != -1? devtolin(a->ac_tty):"?");
```

Comment:
ac_tty is a `dev32_t` aka. `uint32_t` which is presumably why the original code cast -1 to `dev_t`. 
`dev_t` is a `ulong_t` which is of course larger in 64-bit, which is probably what the compiler complained about.

I suggest:
```suggestion
	    a->ac_tty != (dev32_t)-1 ? devtolin(a->ac_tty) : "?");
```